include/appimage/utils/Logger.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <memory>
+#include <functional>
+
+namespace appimage {
+    namespace utils {
+        enum class LogLevel {
+            DEBUG = 0, INFO, WARNING, ERROR
+        };
+
+        typedef std::function<void(const utils::LogLevel& level, const std::string& message)> LogFunctionType;
```

Comment:
Types defined by `typedef` should IMO not be named with the same style as classes. Also, it's a callback, not a log function. The log function is `log()`.

How about `log_callback_t` or so? (Minor issue, though.)

Reply:
done